crates/router/src/connector/tsys/transformers.rs

```diff
@@ -192,12 +192,14 @@ fn get_error_response(
 
 fn get_payments_response(connector_response: TsysResponse) -> types::PaymentsResponseData {
     types::PaymentsResponseData::TransactionResponse {
-        resource_id: types::ResponseId::ConnectorTransactionId(connector_response.transaction_id),
+        resource_id: types::ResponseId::ConnectorTransactionId(
+            connector_response.transaction_id.clone(),
+        ),
         redirection_data: None,
         mandate_reference: None,
         connector_metadata: None,
         network_txn_id: None,
-        connector_response_reference_id: None,
+        connector_response_reference_id: Some(connector_response.transaction_id.clone()),
```

Comment:
```suggestion
        connector_response_reference_id: Some(connector_response.transaction_id),
```

Reply:
This has been implemented now. Thanks for the suggestion.